[AVRO-4133] Support default enum value in Protobuf to Avro

zenfenan opened the pull request:
<!--

*Thank you very much for contributing to Apache Avro - we are happy that you want to help us improve Avro. To help the community review your contribution in the best possible way, please go through the checklist below, which will get the contribution into a shape in which it can be best reviewed.*

*Please understand that we do not do this to make contributions to Avro a hassle. In order to uphold a high standard of quality for code contributions, while at the same time managing a large number of contributions, we need contributors to prepare the contributions well, and give reviewers enough contextual information for the review. Please also understand that contributions that do not follow this guide will take longer to review and thus typically be picked up with lower priority by the community.*

## Contribution Checklist

  - Make sure that the pull request corresponds to a [JIRA issue](https://issues.apache.org/jira/projects/AVRO/issues). Exceptions are made for typos in JavaDoc or documentation files, which need no JIRA issue.
  
  - Name the pull request in the form "AVRO-XXXX: [component] Title of the pull request", where *AVRO-XXXX* should be replaced by the actual issue number. 
    The *component* is optional, but can help identify the correct reviewers faster: either the language ("java", "python") or subsystem such as "build" or "doc" are good candidates.  

  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
  
  - Make sure that the change passes the automated tests. You can [build the entire project](https://github.com/apache/avro/blob/main/BUILD.md) or just the [language-specific SDK](https://avro.apache.org/project/how-to-contribute/#unit-tests).

  - Each pull request should address only one issue, not mix up code from multiple issues.
  
  - Each commit in the pull request has a meaningful commit message (including the JIRA id)

  - Every commit message references Jira issues in their subject lines. In addition, commits follow the guidelines from [How to write a good git commit message](https://chris.beams.io/posts/git-commit/)
    1. Subject is separated from body by a blank line
    1. Subject is limited to 50 characters (not including Jira issue reference)
    1. Subject does not end with a period
    1. Subject uses the imperative mood ("add", not "adding")
    1. Body wraps at 72 characters
    1. Body explains "what" and "why", not "how"

-->

## What is the purpose of the change

*(For example: This pull request improves file read performance by buffering data, fixing AVRO-XXXX.)*
This PR adds the ability to create an enum Schema with the default value when converting Protobuf descriptor to Avro schema.

## Verifying this change

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

## Documentation

- Does this pull request introduce a new feature? (yes)
- If yes, how is the feature documented? (not applicable)

martin-g reviewed (approved):
https://protobuf.dev/programming-guides/proto3/#default - 👍🏻

zenfenan commented:
@martin-g Could you also merge this?

nandorKollar commented:
@zenfenan Would you mind add an extra test case to demonstrate this change? It seems that at the moment nothing tests enum schema conversions, I think `TestProbuf` would be a good place for an extra test case. It appears, that we didn't handle default tags in the past either, like the example in `test.proto`: `optional        A enum     = 16 [default = Z];`. Probably we should raise a separate Jira to handle default tags properly.

opwvhk reviewed (changes requested):
The change itself looks good, but it lacks a test case.
Preferably with the link @martin-g provided: https://protobuf.dev/programming-guides/proto3/#default

zenfenan commented:
@nandorKollar & @opwvhk 

Thanks for your reviews! I realised that you can actually have default value in your protobuf schema until proto2. It is only removed in proto3. Refer: 
- https://protobuf.dev/programming-guides/proto2/#enum-default
- https://protobuf.dev/programming-guides/proto3/#enum-default

So I updated the code to handle that, and also added a test (with new proto3 schema). Let me know if the commits need to be squashed into a single commit.

BTW, the CodeQL violations are on the protobuf generated code. I couldn't find any exclusion config in the project. Let me know if this needs to be handled or not!

zenfenan commented:
@opwvhk I am unable to mark the CodeQL warnings as "Won't fix". I also don't have the permission to merge. Possibly because I am not part of the committers list for this repo. 

If there is nothing else pending, could you resolve the CodeQL warnings and also merge the PR? Thanks!
